### DISCUSSION TITLE
Vite assets in Filament

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#4367] [Help]
I followed the guide in documentation and it works perfectly... almost.

The problem is that assets are available only when I run: `npm run dev` command so Vite server is up. 

I expect when I build assets with command `npm run build`, they are published to /public/build and then without starting Vite server 
assets should be served from this folder, but they are not.

I found this solution:

```php
@production
    @php
        $manifest = json_decode(file_get_contents(public_path('build/manifest.json')), true);
    @endphp
    <script type="module" src="/build/{$manifest['resources/js/app.js']['file']}"></script>
    <link rel="stylesheet" href="/build/{$manifest['resources/js/app.js']['css'][0]}">
@else
    <script type="module" src="http://localhost:3000/@vite/client"></script>
    <script type="module" src="http://localhost:3000/resources/js/app.js"></script>
@endproduction
```

but it requires publishing main layout file and I feel that I am missing something and there is a better way... Is it?

**Comment by danharrin** [comment]
That shouldnt be the case, both commands were tested and worked without overriding files. Can you investigate what the actual problem is?

**Reply by studiowizjo** [reply]
I have no idea what else should I check. Everything is set as in Filament docs. I have not any custom css or js.

Does the APP_ENV setting matter? I tested it with "local" and "production" values, but it doesn`t change anything.
